src/providers/baremetal.rs

```diff
@@ -43,12 +44,18 @@ impl super::MachineSetup for Setup {
 
 impl Setup {
     /// Create a new instance of Setup.
-    pub fn new(
-        addr: impl std::net::ToSocketAddrs,
+    #[instrument(level = "debug")]
+    pub fn new<A: std::net::ToSocketAddrs + std::fmt::Debug>(
+        addr: A,
         username: Option<String>,
-    ) -> Result<Self, Error> {
-        let username: Result<String, Error> = username.map(Ok).unwrap_or_else(|| {
-            let user = String::from_utf8(std::process::Command::new("whoami").output()?.stdout)?;
+    ) -> Result<Self, Report> {
+        let username: Result<String, Report> = username.map(Ok).unwrap_or_else(|| {
+            let stdout = std::process::Command::new("whoami")
+                .output()
+                .wrap_err("whoami to determine local user")?
```

Comment:
this looks like code smell to me, I'd either change this message to represent an actual error or move it into a span that is captured by `tracing-error` when the report is created.

Reply:
Yeah, this one is a little tricky. I don't disagree with you in principle, but this isn't a common pattern in this particular application, and the only thing executed in that span would also be this command. A span or a command wrapper seems like overkill for this one case?

Reply:
Not sure, what do you mean by overkill?

This is what I'm imagining.

```rust
trait Output {
    fn output2(&mut self) -> Result<String, Report>;
}

impl Output for Command {
    #[instrument]
    fn output2(&mut self) -> Result<String, Report> {
        let output = self.output()?;

        let stdout = String::from_utf8_lossy(&output.stdout);

        if !output.status.success() {
            let stderr = String::from_utf8_lossy(&output.stderr);
            eyre!("cmd exited with non-zero status code")
                .add_section(Section { title: "stdout:", body: stdout })
                .add_section(Section { title: "stderr:", body: stderr})
        } else {
            Ok(stdout.into())
        }
    }
}
```

I lean towards it being worth it just because it makes debugging errors from subshells WAY nicer.

Reply:
I guess what I meant is that in this crate, the only place that extension trait would be used would be here for `whoami` — a command that is _very_ unlikely to fail in the first place. Just reporting that "whoami failed" I think is enough context for this particular case. In a program that actually executes more commands though, completely agree this would be useful.

Reply:
oh, I definitely agree with that.